### PR TITLE
Update examples/MNIST Manifest, including Julia 1.10

### LINKED ISSUE #234: Update examples/MNIST
Using latest pkg releases, the example will possibly fail for GPUs because of scalar indexing implicit in the `make2d` function (the use of a colon argument in `reshape`). Either use

```
function make2d(x)
    l = length(x)
    b = size(x)[end]
    reshape(x, div(l, b), b)
end
```

Or `MLUtils.flatten` which does the same thing.

### PR DESCRIPTION
**Pull request by ablaom**
This PR resolves #234. It updates the example/MNIST to use latest package versions and Julia 1.10

To do:

- [x] Test notebook locally with GPU switched on (`acceleration=CUDALibs()`).

cc @EssamWissam

### CODE REVIEW

File: examples/mnist/notebook.jl [hunk 1]
```diff
@@ -77,7 +77,7 @@ function MLJFlux.build(b::MyConvBuilder, rng, n_in, n_out, n_channels)
         MaxPool((2, 2)),
         Conv((k, k), c2 => c3, pad=(p, p), relu, init=init),
```
Comment (EssamWisam) [inline examples/mnist/notebook.jl:78]: Instead of computing the required padding to perserve image size manually above and passing it here, one can just pass the singleton `SamePad()`
https://fluxml.ai/Flux.jl/stable/models/layers/#Convolution-Models

File: examples/mnist/notebook.jl [hunk 2]
```diff
@@ -87,18 +87,19 @@ end
...
-                      epochs=10,
-                      rng=123)
+clf = ImageClassifier(
+    builder=MyConvBuilder(3, 16, 32, 32),
+    batch_size=50,
+    epochs=10,
+    rng=123,
+)
 
 # You can add Flux options `optimiser=...` and `loss=...` here. At
```
Comment (EssamWisam) [inline examples/mnist/notebook.jl:100]:
```suggestion
# You can add to the constructor Flux options `optimiser=...` and `loss=...` here. At
```
because its later mention `To run on a GPU, set acceleration=CUDALib() and omit rng.` which if followed by a call to `machine` (so one may think they should add it the machine and not the constructor if that's not explicit.)

File: examples/mnist/notebook.jl [hunk 3]
```diff
@@ -129,7 +130,7 @@ fit!(mach, rows=1:500);
 # Computing an out-of-sample estimate of the loss:
 
 predicted_labels = predict(mach, rows=501:1000);
-cross_entropy(predicted_labels, labels[501:1000]) |> mean
+cross_entropy(predicted_labels, labels[501:1000])
```
Comment (EssamWisam) [inline examples/mnist/notebook.jl:133]: I do personally prefer training and testing data to be separate data structures (or at least to make this more generic by deciding a split size initially that could be changed as needed).

It may confuse some that we are loading the training data of MNIST then using the first 500 examples for actual training and the second 500 for testing/validation.

File: examples/mnist/notebook.jl [hunk 4]
```diff
@@ -129,7 +130,7 @@ fit!(mach, rows=1:500);
...
 
 # Or, in one line:
```
Comment (EssamWisam) [inline examples/mnist/notebook.jl:135]:
```suggestion
# Or to fit and predict, in one line:
```

File: examples/mnist/notebook.jl [hunk 5]
```diff
@@ -63,8 +65,6 @@ struct MyConvBuilder
     channels3::Int
 end
 
-make2d(x::AbstractArray) = reshape(x, :, size(x)[end])
```
Comment (EssamWisam) [inline examples/mnist/notebook.jl:66]: There is another instance of this below around line 173, should it be also replaced with `MLUtils.flatten`?

File: examples/mnist/notebook.jl [hunk 6]
```diff
@@ -239,18 +246,21 @@ savefig(joinpath(DIR, "loss.png"))
...
-     xlab = "epoch")
+plot(
+    epochs,
+    parameter_means2,
+    title="Flux parameter mean weights",
+    xlab = "epoch",
+)
 
-# **Note.** The the higher the number, the deeper the chain parameter.
+# **Note.** The higher the number, the deeper the chain parameter.
```
Comment (EssamWisam) [inline examples/mnist/notebook.jl:256]: I don't get in what sense its deeper because perhaps this is using the same number of hidden layers regardless to epochs.

File: examples/mnist/notebook.jl (hunk 3 above)
Comment (ablaom) [reply examples/mnist/notebook.jl:133]: Addressed.

File: examples/mnist/notebook.jl (hunk 2 above)
Comment (ablaom) [reply examples/mnist/notebook.jl:100]: Addressed in a new commit.

File: examples/mnist/notebook.jl (hunk 1 above)
Comment (ablaom) [reply examples/mnist/notebook.jl:78]: Good to know. I will leave it as is, for less "hidden knowledge" for begginners, but will add your alternative in the form of a comment.

File: examples/mnist/notebook.jl (hunk 6 above)
Comment (ablaom) [reply examples/mnist/notebook.jl:256]: We are reporting the mean value of weights for different layers. The higher the index, the deeper the layer we are averaging. 

Added a comment to clarify, but let me know if this is still not clear or if you believe my assertion is not correct.

File: examples/mnist/notebook.jl (hunk 5 above)
Comment (ablaom) [reply examples/mnist/notebook.jl:66]: Okay, it looks like `make2d` is not used, and we can use `vec` for `make1d`. So I've done that.

**Comment by ablaom** [comment]
OKay @EssamWisam I'm ready for you to have another look at this.

File: examples/mnist/notebook.jl (hunk 6 above)
Comment (EssamWisam) [reply examples/mnist/notebook.jl]: OK. I hadn't seen the plot or noticed the `div` operation when I commented. Now I understand that the higher the index in the means list, the deeper the layer. I think I initially understood "number" to be the epoch number.

Those we can either just assume users will conclude that "number" is the index in the plot or clarify it.

**Comment by EssamWisam** [comment]
> OKay @EssamWisam I'm ready for you to have another look at this.

Redefines perfection 👌. I think we are ready to merge this just after we resolve anything, if needed, in my comment above.

**Comment by EssamWisam** [comment]
Oh and also after exposing it in the documentation: obvious by moving the tutorial to the `docs` folder and then putting the path in `make.jl` (in the commented tutorials section).

**Comment by ablaom** [comment]
> I think we are ready to merge this just after we resolve anything, if needed, in my comment above.

Sorry, which comment is that?

> Oh and also after exposing it in the documentation: obvious by moving the tutorial to the docs folder and then putting the path in make.jl (in the commented tutorials section).

Mmm. I'm going to postpone that step, but thanks for flagging.

File: examples/mnist/notebook.jl (hunk 6 above)
Comment (EssamWisam) [reply examples/mnist/notebook.jl]: @ablaom this comment.
